CMakeLists.txt

```diff
@@ -101,12 +133,39 @@ foreach(irontarget ${IRON_TARGETS})
         "${CMAKE_CURRENT_SOURCE_DIR}/src"
         ${CMAKE_CURRENT_BINARY_DIR}
     )
+    if(WORKING_REAL_PRECISION_SINGLE)
+        target_compile_definitions(${irontarget} PUBLIC SINGLE_REAL_PRECISION)
+    else()
+        target_compile_definitions(${irontarget} PUBLIC DOUBLE_REAL_PRECISION)
+    endif()
+    if (IS_DEBUG_BUILD)
+        target_compile_definitions(${irontarget} PUBLIC DEBUG)
+    endif()
     if (WITH_CELLML)
-        target_compile_definitions(${irontarget} PUBLIC USECELLML)
+        target_compile_definitions(${irontarget} PUBLIC WITH_CELLML)
     endif()
     if (WITH_FIELDML)
-        target_compile_definitions(${irontarget} PUBLIC USEFIELDML)
+        target_compile_definitions(${irontarget} PUBLIC WITH_FIELDML)
     endif()
+    if (WITH_HYPRE)
+        target_compile_definitions(${irontarget} PUBLIC WITH_HYPRE)
+    endif()
+    if (WITH_SUNDIALS)
+        target_compile_definitions(${irontarget} PUBLIC WITH_SUNDIALS)
+    endif()
+    if (WITH_MUMPS)
+        target_compile_definitions(${irontarget} PUBLIC WITH_MUMPS)
+    endif()
+    if (WITH_SCALAPACK)
+        target_compile_definitions(${irontarget} PUBLIC WITH_SCALAPACK)
+    endif()
+    if (WITH_PETSC)
+        target_compile_definitions(${irontarget} PUBLIC WITH_PETSC)
+    endif()
+    if (WITH_DIAGNOSTICS)
+        target_compile_definitions(${irontarget} PUBLIC WITH_DIAGNOSTICS)
+    endif()
+    
```

Comment:
Maybe, but this is pretty standard and makes it explicit when it compiles. Maybe we can think about a header file later if/when there are more options.